discovery/eureka/client.go

```diff
@@ -0,0 +1,131 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eureka
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type Applications struct {
+	VersionsDelta int           `xml:"versions__delta"`
+	AppsHashcode  string        `xml:"apps__hashcode"`
+	Applications  []Application `xml:"application,omitempty"`
+}
+
+type Application struct {
+	Name      string     `xml:"name"`
+	Instances []Instance `xml:"instance"`
+}
+
+type Port struct {
+	Port    int  `xml:",chardata"`
+	Enabled bool `xml:"enabled,attr"`
+}
+
+type Instance struct {
+	HostName                      string          `xml:"hostName"`
+	HomePageURL                   string          `xml:"homePageUrl,omitempty"`
+	StatusPageURL                 string          `xml:"statusPageUrl"`
+	HealthCheckURL                string          `xml:"healthCheckUrl,omitempty"`
+	App                           string          `xml:"app"`
+	IPAddr                        string          `xml:"ipAddr"`
+	VipAddress                    string          `xml:"vipAddress"`
+	SecureVipAddress              string          `xml:"secureVipAddress,omitempty"`
+	Status                        string          `xml:"status"`
+	Port                          *Port           `xml:"port,omitempty"`
+	SecurePort                    *Port           `xml:"securePort,omitempty"`
+	DataCenterInfo                *DataCenterInfo `xml:"dataCenterInfo"`
+	LeaseInfo                     *LeaseInfo      `xml:"leaseInfo,omitempty"`
+	Metadata                      *MetaData       `xml:"metadata,omitempty"`
+	IsCoordinatingDiscoveryServer bool            `xml:"isCoordinatingDiscoveryServer,omitempty"`
+	LastUpdatedTimestamp          int             `xml:"lastUpdatedTimestamp,omitempty"`
+	LastDirtyTimestamp            int             `xml:"lastDirtyTimestamp,omitempty"`
+	ActionType                    string          `xml:"actionType,omitempty"`
+	Overriddenstatus              string          `xml:"overriddenstatus,omitempty"`
+	CountryID                     int             `xml:"countryId,omitempty"`
+	InstanceID                    string          `xml:"instanceId,omitempty"`
+}
+
+type DataCenterInfo struct {
+	Name     string              `xml:"name"`
+	Class    string              `xml:"class,attr"`
+	Metadata *DataCenterMetadata `xml:"metadata,omitempty"`
+}
+
+type DataCenterMetadata struct {
+	AmiLaunchIndex   string `xml:"ami-launch-index,omitempty"`
```

Comment:
Why is there EC2-specific stuff in a non-EC2 SD?

Is this actually a first-class thing in Eureka, or is this just an arbitrary map? Looking at the code, it's a map - so should be treated as-such.

Reply:
Indeed, and it is also optional: https://github.com/Netflix/eureka/wiki/Eureka-REST-operations

Reply:
Those fields should not me enumarated. It should be a map.

Reply:
I'll change it to a map.